usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -577,3 +586,91 @@ def fetch_naics_hierarchy(naics: str) -> dict:
         pass
     results = {"toptier_code": toptier_code, "midtier_code": midtier_code, "base_code": base_code}
     return results
+
+
+def fetch_account_details_award(award_id: int) -> dict:
+    queryset = (
+        FinancialAccountsByAwards.objects.filter(award_id=award_id)
+        .values("disaster_emergency_fund")
+        .annotate(
+            obligated_amount=Sum("obligations_delivered_orders_unpaid_total_cpe"),
+            gross_outlay_amount=Sum("gross_outlay_amount_by_award_cpe"),
+        )
+    )
+    outlay_by_code = []
+    obligation_by_code = []
+    total_outlay = 0
+    total_obligations = 0
+    for row in queryset:
+        total_outlay += row["gross_outlay_amount"] if row["gross_outlay_amount"] is not None else 0
+        total_obligations += row["obligated_amount"] if row["obligated_amount"] is not None else 0
+        if row["disaster_emergency_fund"] is not None:
+            outlay_by_code.append({"code": row["disaster_emergency_fund"], "amount": row["gross_outlay_amount"]})
+            obligation_by_code.append({"code": row["disaster_emergency_fund"], "amount": row["obligated_amount"]})
+
+    results = {
+        "total_account_outlay": total_outlay,
+        "total_account_obligation": total_obligations,
+        "account_outlays": outlay_by_code,
+        "account_obligations": obligation_by_code,
+    }
+    return results
+
+
+def fetch_account_details_idv(award_id: int) -> dict:
+
+    child_award_sql = """
+    select
+        ac.id as award_id
+    from
+        parent_award pap
+        inner join awards ap on ap.id = pap.award_id
+        inner join awards ac on ac.fpds_parent_agency_id = ap.fpds_agency_id and ac.parent_award_piid = ap.piid
+    where
+        pap.award_id = {award_id}
+    """
+    grandchild_award_sql = """
+    select
+        ac.id as award_id
+    from
+        parent_award pap
+        inner join parent_award pac on pac.parent_award_id = pap.award_id
+        inner join awards ap on ap.id = pac.award_id
+        inner join awards ac on ac.fpds_parent_agency_id = ap.fpds_agency_id and ac.parent_award_piid = ap.piid and
+            ac.type not like 'IDV%'
+
+    where
+        pap.award_id = {award_id}
+    """
+    children = execute_sql_to_ordered_dictionary(child_award_sql.format(award_id=award_id))
+    grandchildren = execute_sql_to_ordered_dictionary(grandchild_award_sql.format(award_id=award_id))
+    award_ids = []
+    award_ids.extend([x["award_id"] for x in children])
+    award_ids.extend([x["award_id"] for x in grandchildren])
+    award_ids.append(award_id)
```

Comment:
_IF_ we're not supposed to be summing IDV values, then we also need to not append `award_id`.